packages/library-base/src/commonMain/kotlin/io/realm/RealmQuery.kt

```diff
@@ -0,0 +1,361 @@
+     *
+     * The change calculations will run on the thread represented by
+     * [RealmConfiguration.Builder.notificationDispatcher].
+     *
+     * @return a flow representing changes to the [RealmObject] resulting from running this query.
+     */
+    override fun asFlow(): Flow<E?>
+}
+
+/**
+ * Queries that return scalar values. This type of query is used to more accurately represent the
+ * results provided by some query operations, e.g. [RealmQuery.count] or [RealmQuery.min].
+ */
+interface RealmScalarQuery<E> : Flowable<E> {
+    /**
+     * Returns the value of a scalar query as an [E] in a blocking fashion. The result may be `null`
+     * depending on the type of scalar query:
+     *
+     * - `[min]`, `[max]` return [E] or `null` if no objects are present
+     * - `[count]` returns [Long]
+     *
+     * Launching heavy queries from the UI thread may result in a drop of frames or even ANRs. **We
+     * do not recommend doing so.** If you want to prevent these behaviors you can obtain the
+     * values asynchronously using [asFlow] instead.
+     *
+     * @return an [E] containing the result of the scalar query or `null` depending on the query
+     * being executed.
+     */
+    fun find(): E?
+
+    /**
+     * Calculates the value that fulfills the query conditions and returns it asynchronously as a
+     * [Flow].
+     *
+     * If there is any changes to the objects represented by the query backing the value, the flow
+     * will emit the updated value. The flow will continue running indefinitely until canceled.
+     *
+     * The change calculations will run on the thread represented by
+     * [RealmConfiguration.Builder.notificationDispatcher].
+     *
+     * @return a flow representing changes to the [RealmResults] resulting from running this query.
+     */
+    override fun asFlow(): Flow<E?>
+}
+
+/**
+ * This enum describes the sorting order used in Realm queries.
+ *
+ * @see [RealmQuery.sort]
+ */
+enum class QuerySort {
+    ASCENDING,
+    DESCENDING
+}
+
+/**
+ * Similar to [RealmQuery.min] but the type parameter is automatically inferred.
+ */
+inline fun <reified T : Any> RealmQuery<*>.min(property: String): RealmScalarQuery<T> =
+    min(property, T::class)
+
+/**
+ * Similar to [RealmQuery.max] but the type parameter is automatically inferred.
+ */
+inline fun <reified T : Any> RealmQuery<*>.max(property: String): RealmScalarQuery<T> =
+    max(property, T::class)
+
+/**
+ * Similar to [RealmQuery.sum] but the type parameter is automatically inferred.
+ */
+inline fun <reified T : Any> RealmQuery<*>.sum(property: String): RealmScalarQuery<T> =
+    sum(property, T::class)
+
+/**
+ * Similar to [RealmQuery.find] but it receives a block in which the [RealmResults] from the query
+ * are provided.
+ */
+fun <T> RealmQuery<T>.find(block: (RealmResults<T>) -> Unit): RealmResults<T> = find().let {
+    block.invoke(it)
+    it
```

Comment:
Shouldn't we return whatever people return from the closure instead? Or am I missing a use case here? Same for the others.